packages/backend-app-api/src/wiring/ServiceRegistry.ts

```diff
@@ -146,6 +146,43 @@ export class ServiceRegistry implements EnumerableServiceHolder {
     }
   }
 
+  #checkForCircularDeps(factory: InternalServiceFactory, pluginId: string) {
+    const head = factory;
+
+    const nodes = Object.values(factory.deps);
+
+    const depChain: Array<ServiceRef<unknown, 'plugin' | 'root'>> = [
+      head.service,
+    ];
+
+    let node = nodes.shift();
+
+    if (node) {
+      depChain.push(node);
+    }
+
+    while (!!node && node.id !== head.service.id) {
```

Comment:
You'll want to keep track of visited nodes, or you might end up with an infinite loop in the case where the starting node is not part of the cycle, for example `A -> B -> C -> B`.

Reply:
Added a unit test to cover this scenario